client/src/components/App/Content/LandingPage/adminLandingPage/investigationsInfo/investigationsInfo.tsx

```diff
@@ -1,50 +1,91 @@
-import React from 'react';
-import { useHistory } from 'react-router-dom';
+import React, { useMemo } from 'react';
 import NavigateBeforeIcon from '@material-ui/icons/NavigateBefore';
-import { Card, CardContent, Grid, IconButton, Typography } from '@material-ui/core';
+import { CardContent, Grid, IconButton, Typography } from '@material-ui/core';
 
-import { landingPageRoute } from 'Utils/Routes/Routes';
 import InvestigationChart from 'models/InvestigationChart';
+import FilterRulesVariables from 'models/FilterRulesVariables';
+import statusToFilterConvertor from 'commons/statusToFilterConvertor';
+import FilterRulesDescription from 'models/enums/FilterRulesDescription';
+import { InvesitgationInfoStatistics } from 'models/InvestigationStatistics';
 
-import useStyles from './investigationsInfoStyles';
+import LoadingCard from '../LoadingCard/LoadingCard';
+import useStyles, { cardWidth, cardHeight } from './investigationsInfoStyles';
 import InvestigationBarChart from './InvestigationBarChart/InvestigationBarChart';
-import InvestigationChartData from './InvestigationBarChart/InvestigationChartData';
 import InvestigationInfoButton from './investigationInfoButton/investigationInfoButton';
 
-const InvestigationsInfo: React.FC = (): JSX.Element => {
+const convertorsToGraph: { [T in keyof InvesitgationInfoStatistics]: Omit<InvestigationChart, 'value'>} = {
+    newInvestigations: {
+        id: FilterRulesDescription.NEW,
+        color: '#1F78B4'
+    },
+    inProcessInvestigations: {
+        id: FilterRulesDescription.IN_PROCESS,
+        color: 'grey'
+    },
+    unassignedInvestigations: {
+        id: FilterRulesDescription.UNASSIGNED,
+        color: '#33A02C'
+    },
+    inactiveInvestigations: {
+        id: FilterRulesDescription.INACTIVE,
+        color: '#F95959'
+    }
+}
+
+const InvestigationsInfo: React.FC<Props> = (props: Props): JSX.Element => {
     const classes = useStyles();
 
-    let history = useHistory();
+    const { onInfoButtonClick, investigationsStatistics, allInvestigationsCount, isLoading } = props;
+
+    const investigationsGraphData = useMemo<InvestigationChart[]>(() => {
```

Comment:
maybe add type to avoid as keyof typeof convertion in line 65

Reply:
fixed

Reply:
is the InvestigationChart id property necessarily keyof typeof statusToFilterConvertor? (currently string)
if it is, it could be changed to  keyof typeof statusToFilterConvertor in the InvestigationChart interface itself

Reply:
fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1052/commits/3e22cee6e0d949dd086c5002c57ba1b1c2d4361d)